chia/wallet/trade_manager.py

```diff
@@ -101,19 +101,12 @@ async def get_coins_of_interest(
         Returns list of coins we want to check if they are included in filter,
         These will include coins that belong to us and coins that that on other side of treade
         """
-        all_pending = []
-        pending_accept = await self.get_offers_with_status(TradeStatus.PENDING_ACCEPT)
-        pending_confirm = await self.get_offers_with_status(TradeStatus.PENDING_CONFIRM)
-        pending_cancel = await self.get_offers_with_status(TradeStatus.PENDING_CANCEL)
-        all_pending.extend(pending_accept)
-        all_pending.extend(pending_confirm)
-        all_pending.extend(pending_cancel)
+        coins = await self.trade_store.get_coins_of_interest_with_trade_statuses(
+            trade_statuses=[TradeStatus.PENDING_ACCEPT, TradeStatus.PENDING_CONFIRM, TradeStatus.PENDING_CANCEL]
+        )
         interested_dict = {}
-
-        for trade in all_pending:
-            for coin in trade.coins_of_interest:
-                interested_dict[coin.name()] = coin
-
+        for coin in coins:
+            interested_dict[coin.name()] = coin
```

Comment:
the only thing we do with the coins is to call `name()`. we could avoid a DB join and this call

Reply:
but `coin` is still needed as value to the dict? or are you saying doing another select here to fetch coins from coin_ids?

Reply:
oops. I missed that :)

Reply:
actually.. as far as I can tell, this function could return a set of coin_ids. there are only two callers of it, and both of them only care about the coin_id (afaict)